Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -0,0 +1,332 @@
+#include <API/EditorAssetSystemAPI.h>
+#include <AzCore/IO/FileIO.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/Prefab/PrefabLoaderInterface.h>
+#include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
+#include <AzToolsFramework/ToolsComponents/GenericComponentWrapper.h>
+#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
+
+namespace ROS2
+{
+    URDFPrefabMaker::URDFPrefabMaker(
+        const AZStd::string& modelFilePath,
+        urdf::ModelInterfaceSharedPtr model,
+        AZStd::string prefabPath,
+        const AZStd::shared_ptr<Utils::UrdfAssetMap> urdfAssetsMapping)
+        : m_model(model)
+        , m_visualsMaker(model->materials_, urdfAssetsMapping)
+        , m_collidersMaker(urdfAssetsMapping)
+        , m_prefabPath(std::move(prefabPath))
+        , m_urdfAssetsMapping(urdfAssetsMapping)
+    {
+        AZ_Assert(!m_prefabPath.empty(), "Prefab path is empty");
+        AZ_Assert(m_model, "Model is nullptr");
+    }
+
+    void URDFPrefabMaker::LoadURDF(BuildReadyCallback buildReadyCb)
+    {
+        m_notifyBuildReadyCb = buildReadyCb;
+
+        // Request the build of collider meshes by constructing .assetinfo files.
+        BuildAssetsForLink(m_model->root_link_);
+
+        // Wait for all collider meshes to be ready
+        m_collidersMaker.ProcessMeshes(buildReadyCb);
+    }
+
+    void URDFPrefabMaker::BuildAssetsForLink(urdf::LinkSharedPtr link)
+    {
+        m_collidersMaker.BuildColliders(link);
+        for (auto childLink : link->child_links)
+        {
+            BuildAssetsForLink(childLink);
+        }
+    }
+
+    AzToolsFramework::Prefab::CreatePrefabResult URDFPrefabMaker::CreatePrefabFromURDF()
+    {
+        {
+            AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
+            m_status.clear();
+        }
+        // TODO - this is PoC code, restructure when developing semantics of URDF->Prefab/Entities/Components mapping
+        AZStd::unordered_map<AZStd::string, AzToolsFramework::Prefab::PrefabEntityResult> created_links;
+        AzToolsFramework::Prefab::PrefabEntityResult createEntityRoot = AddEntitiesForLink(m_model->root_link_, AZ::EntityId());
+        AZStd::string root_name(m_model->root_link_->name.c_str(), m_model->root_link_->name.size());
+        created_links[root_name] = createEntityRoot;
+        if (!createEntityRoot.IsSuccess())
+        {
+            return AZ::Failure(AZStd::string(createEntityRoot.GetError()));
+        }
+
+        auto links = Utils::GetAllLinks(m_model->root_link_->child_links);
+
+        // create links
+        for (const auto& [name, link_ptr] : links)
+        {
+            created_links[name] = AddEntitiesForLink(link_ptr, createEntityRoot.GetValue());
+        }
+
+        for (const auto& [name, result] : created_links)
+        {
+            AZ_TracePrintf(
+                "CreatePrefabFromURDF",
+                "Link with name %s was created as: %s",
```

Comment:
This `AZ_TracePrintf` needs a newline at the end of the format string.
```suggestion
                "Link with name %s was created as: %s\n",
```